robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -318,14 +331,19 @@ def __init__(self, robot, item, arm):
         check_type(arm, Arm)
 
         with self:
-            smach.StateMachine.add('PREPARE_GRASP', PrepareEdGrasp(robot, arm, item),
-                                   transitions={'succeeded': 'NAVIGATE_TO_GRAB',
-                                                'failed': 'RESET_FAILURE'})
-
             smach.StateMachine.add('NAVIGATE_TO_GRAB', NavigateToGrasp(robot, item, arm),
                                    transitions={'unreachable': 'RESET_FAILURE',
                                                 'goal_not_defined': 'RESET_FAILURE',
-                                                'arrived': 'GRAB'})
+                                                'arrived': 'PREPARE_GRASP'})
+
+            smach.StateMachine.add('PREPARE_GRASP', PrepareEdGrasp(robot, arm, item),
+                                   transitions={'succeeded': 'GRAB',
+                                                'failed': 'RESET_FAILURE'})
+
+#            smach.StateMachine.add('NAVIGATE_TO_GRAB', NavigateToGrasp(robot, item, arm),
+#                                   transitions={'unreachable': 'RESET_FAILURE',
+#                                                'goal_not_defined': 'RESET_FAILURE',
+#                                                'arrived': 'GRAB'})
```

Comment:
We can keep these changes, it is necessary for Hero to navigate before going to the pre-grasp pose since the updating of the to grasp entity occurs after the navigating to grasp. So if Hero goes to the pre-grasp pose before updating the entity, he can't 'see' the entity. For Amigo this is not really a problem, so keeping these changes would be best.